### PR TITLE
Reorganize coverage report for set session description

### PR DESCRIPTION
**Pull request by soareschen**
This is another attempt for me to organize the coverage information for the test files.

It has been very difficult to track what has been tested and not tested for webrtc-pc, especially for section 4.3.1.6. Set the RTCSessionSessionDescription. The section contains a large number of steps across multiple domains, making it impossible to test all logic in a single test files. Instead, the logic for 4.3.16. is tested across many test files that test particular methods.

This creates problem when trying to relate the tests and the spec, because there is no central reference point for where are the tests for each step. The current manual approach has been to read the content for each step, guess which test file might cover that step, and go into the particular test file to search for the test. This can slow down productivity significantly when things need to be updated routinely.

To solve this problem, I created a new coverage/ directory and have a text file that annotate whether a step has been covered, and which test file cover which step. This should make it easy to keep track of the tests.

The coverage report is currently has adhoc annotation and cannot be parsed easily, but it gets the job done for now. In the future if this approach is working well, we can transition to a format that is both machine and human friendly so that aggregated coverage report can be generated automatically.

### CODE REVIEW

File: webrtc/RTCPeerConnection-setLocalDescription.html
Comment (rwaldron) [inline webrtc/RTCPeerConnection-setLocalDescription.html]: 4.4.2 => 4.3.2

**Comment by rwaldron** [comment]
A lot of this is pretty clearly marked at this point, for example: https://rwaldron.github.io/webrtc-pc/#dom-rtcpeerconnection-setlocaldescription needs one step tested and will be complete

File: webrtc/coverage/set-session-description.txt
Comment (rwaldron) [inline webrtc/coverage/set-session-description.txt]: 4.4.1.6 => 4.3.1.6

File: webrtc/coverage/set-session-description.txt
Comment (rwaldron) [inline webrtc/coverage/set-session-description.txt]: This is not the contents of 2.1.2 that appear in the link you gave. This is step 2.1.2: 


> If elements of the SDP were modified, then reject p with a newly created InvalidModificationError and abort these steps.

File: webrtc/coverage/set-session-description.txt
Comment (rwaldron) [inline webrtc/coverage/set-session-description.txt]: This is the content of 2.1.5. The contents of 2.1.4 are incorrectly shown above in 2.1.3.

File: webrtc/coverage/set-session-description.txt
Comment (rwaldron) [inline webrtc/coverage/set-session-description.txt]: This is the content for 2.1.4

File: webrtc/coverage/set-session-description.txt
Comment (rwaldron) [inline webrtc/coverage/set-session-description.txt]: Most of this matches my records (the things yours marking "TODO" are things I had marked "untested", but I've updated my coverage spec to match your "TODO")

File: webrtc/coverage/set-session-description.txt
Comment (rwaldron) [inline webrtc/coverage/set-session-description.txt]: I've updated the coverage spec to make these "untestable" instead of "untested"

File: webrtc/coverage/set-session-description.txt
Comment (rwaldron) [inline webrtc/coverage/set-session-description.txt]: The coverage spec had all of these marked as "untested", they are now "todo"

File: webrtc/RTCPeerConnection-setLocalDescription.html
Comment (rwaldron) [inline webrtc/RTCPeerConnection-setLocalDescription.html]: 4.4.2 => 4.3.2

File: webrtc/RTCPeerConnection-setLocalDescription.html
Comment (rwaldron) [inline webrtc/RTCPeerConnection-setLocalDescription.html]: 4.4.2 => 4.3.2

**Review by rwaldron — CHANGES_REQUESTED** [review]

File: webrtc/coverage/set-session-description.txt
Comment (soareschen) [inline webrtc/coverage/set-session-description.txt]: Ugh, I must have messed up and copied from the tip of tree draft.

File: webrtc/coverage/set-session-description.txt
Comment (soareschen) [inline webrtc/coverage/set-session-description.txt]: Yeah I am kind of loose between the terms "not tested", "untested", and "todo", though they all mean the same. "untested" is kind of look too similar to "untestable", "not tested" is consist of two words, and "todo" is kind of short more obvious on what still need to be tested.

**Comment by soareschen** [comment]
> A lot of this is pretty clearly marked at this point

@rwaldron yes they are marked at your annotated spec, but it doesn't specify which test file is associated with the corresponding steps. So while it is easy to browse through and get a rough idea of the coverage status, it is difficult for development tasks such as updating the spec, and finding the tests.

We still need to figure ways to manage the synchronization between the specs, tests, and coverage. As new editor's draft coming, there need to be ways to partially update the tests and coverage status in separate PRs, instead of doing it in one big update. 

Unless we have infinite time maintaining the tests, I expect there will be a point of time when different test files reference to different versions of editor's drafts. Therefore it is important that future readers/maintainers (which may or may not be ourselves) can get a quick glance of which version of spec fragments was being referenced when the test was written and the coverage status. And ideally they shouldn't have to scan through the entire editor's draft of that time to figure that out.

Hope this explains the annotation approaches that I am trying to come out with.

File: webrtc/RTCPeerConnection-ontrack.html
Comment (soareschen) [inline webrtc/RTCPeerConnection-ontrack.html]: I haven't really read through in detail but this is one of the old tests written by @fippo. Currently just moving it as is to a more appropriate `ontrack` test file. Planning to study and write detailed tests for `ontrack` soon in another PR.

**Comment by soareschen** [comment]
I have 2 more branches in the pipeline that depend on this PR. 

[The first one](https://github.com/soareschen/web-platform-tests/compare/set-session-description-2...soareschen:set-session-description-4) splits the test files into multiple smaller ones, e.g. RTCPeerConnection-setLocalDescription-answer.html, in attempt to manage the sanity of the test files become too large and cluttered with many unordered tests.

[The second one](https://github.com/soareschen/web-platform-tests/compare/set-session-description-4...soareschen:set-session-description-5) adds new tests for pranswer and rollback. Technically the addition doesn't really depend on these reorganizations. But I had problem keeping track of what still needs to be tested, and it is only after these reorganizations that I am able to gain back my sanity of adding new tests to set*Description.

Similarly the future plan is to add tests for the remaining parts, e.g. effects on transceivers, on separate test files after these branches are merged.

The other branches still need to have some post-split polishing. And to keep them easier to review, I'd on hold from creating them as new PRs until this is merged.

File: webrtc/RTCPeerConnection-ontrack.html
Comment (rwaldron) [inline webrtc/RTCPeerConnection-ontrack.html]: Oh, I see. We'll push it through then

**Review by rwaldron — APPROVED** [review]

**Pull request merged**
